gremlin-javascript/src/main/javascript/gremlin-javascript/lib/utils.js

```diff
@@ -90,10 +90,24 @@ function generateUserAgent() {
   } catch (e) {
     driverVersion = 'NotAvailable';
   }
-  const runtimeVersion = process.version.replace(' ', '_');
-  const osName = os.platform().replace(' ', '_');
-  const osVersion = os.release().replace(' ', '_');
-  const cpuArch = process.arch.replace(' ', '_');
+  let runtimeVersion;
+  let osName;
+  let osVersion;
+  let cpuArch;
+  runtimeVersion = osName = osVersion = cpuArch = 'NotAvailable';
+  if (process != null) {
+    if (process.version != null) {
```

Comment:
Can `version` be undefined?
```suggestion
    if (process.version) {
```
same question for `arch`

Reply:
Yes, it makes sense. If the process is not null or undefined, then it must have a version and arch